torch/csrc/jit/passes/onnx/remove_inplace_ops_for_onnx.cpp

```diff
@@ -535,7 +543,10 @@ void RegisterInplaceNodeInBlocks(
       next_node,
       orig_data->debugName());
 
-  MatchIfBlocksOutputForValue(orig_data, outer_block, new_inplace_node);
+  while (nullptr != outer_block->owningNode()) {
```

Comment:
for nested if blocks, do we need to propagate all the way to the root? i guess it makes sense only propagate to the same level of block owning node of orig_data. I see this is an old logic first appeared in the register in loop blocks for index_put, it may need update there as well.

Reply:
Thanks for mentioning this, will do.